producer.py

```diff
@@ -120,34 +124,36 @@ async def main() -> None:
     si_unanalyzed_package = SIUnanalyzedPackageMessage()
 
     for package_name, package_version, index_url, solver_name in packages:
-        for index_url in [index_url] if index_url is not None else indexes:
-            try:
-                async_tasks.append(
-                    unresolved_package.publish_to_topic(
-                        unresolved_package.MessageContents(
-                            package_name=package_name,
-                            package_version=package_version,
-                            index_url=[index_url],
-                            solver=solver_name,
-                            component_name=COMPONENT_NAME,
-                            service_version=__service_version__,
+        if THOTH_GRAPH_REFRESH_SOLVER:
+            for index_url in [index_url] if index_url is not None else indexes:
+                try:
+                    async_tasks.append(
+                        unresolved_package.publish_to_topic(
+                            unresolved_package.MessageContents(
+                                package_name=package_name,
+                                package_version=package_version,
+                                index_url=[index_url],
+                                solver=solver_name,
+                                component_name=COMPONENT_NAME,
+                                service_version=__service_version__,
+                            )
                         )
                     )
-                )
-                _LOGGER.info(
-                    "Published message for solver %r for package %r in version %r from index %r",
-                    solver_name,
-                    package_name,
-                    package_version,
-                    index_url,
-                )
-            except Exception as identifier:
-                _LOGGER.exception(
-                    "Failed to publish solver message with the following error message: %r",
-                    identifier,
-                )
+                    _LOGGER.info(
+                        "Published message for solver %r for package %r in version %r from index %r",
+                        solver_name,
+                        package_name,
+                        package_version,
+                        index_url,
+                    )
+                except Exception as identifier:
+                    _LOGGER.exception(
+                        "Failed to publish solver message with the following error message: %r",
+                        identifier,
+                    )
 
-            # Send reverse solver message if not done for this packge, package_version
+        # Send reverse solver message if not done for this packge, package_version
+        if THOTH_GRAPH_REFRESH_REVSOLVER:
```

Comment:
Does this block need tabbed over?

Reply:
I don't thinl the `index_url` or `solver_name` so I thought we have the package_name and package_version from above so I tabbed it over so that resolver messages can be enabled and solver be disabled, any suggestions?

Reply:
I think if both THOTH_GRAPH_REFRESH_SOLVER and THOTH_GRAPH_REFRESH_REVSOLVER are set to 0, the check should be done before querying the database, so we avoid the call. WDYT?

Reply:
You are right, I will add that thanks!